src/libcore/iter.rs

```diff
@@ -1547,23 +1626,164 @@ impl<I: Iterator> IntoIterator for I {
     }
 }
 
-/// A type growable from an `Iterator` implementation
+/// Extend a collection with the contents of an iterator.
+///
+/// Iterators produce a series of values, and collections can also be though of
+/// a series of values. The `Extend` trait bridges this gap, allowing you to
+/// extend a collection by including the contents of that iterator.
+/// 
+/// # Examples
+///
+/// Basic usage:
+///
+/// ```
+/// // You can extend a String with some chars:
+/// let mut message = String::from("The first three letters are: ");
+///
+/// message.extend(['a', 'b', 'c'].iter());
```

Comment:
I think this may even be able to be `&['a', 'b', 'c']` as `&[T; 3]` implements `IntoIterator`